content/docs/introduction/comparison.md

```diff
@@ -160,6 +197,95 @@ simple aggregations via its API.
-exceeded.
+Prometheus will be simpler to run initially, but will require explicit sharding
+once the capacity of a single node is exceeded.
+
+### Summary
+
+Prometheus offers a much richer query language, can handle higher cardinality
+metrics and forms part of a complete monitoring system. If you're already
+running Hadoop and value long term storage over these benefits, OpenTSDB is a
+good choice.
+
+
+## Prometheus vs. Nagios
+
+[Nagios](https://www.nagios.org/) is a monitoring system that originated in the
+90s as NetSaint.
+
+### Scope
+
+Nagios is primarily about alerting based on the exit codes of scripts. These are called "checks".
+There is silencing of individual alerts, however no grouping, routing or deduplication.
+
+There are a variety of plugins. For example, piping the few kilobytes of
+perfData plugins are allowed to return [to a time series database such as
+Graphite](https://github.com/shawn-sterling/graphios) or using NRPE to [run
+checks on remote
+machines](https://exchange.nagios.org/directory/Addons/Monitoring-Agents/NRPE--2D-Nagios-Remote-Plugin-Executor/details).
+
+### Data model
+
+Nagios is host-based. Each host can have one or more services, which has one check.
+
+There is no notion of labels or a query language.
+
+### Storage
+
+Nagios has no storage per-se, beyond the current check state.
+There are plugins which can store data such as [for visualisation](https://docs.pnp4nagios.org/).
+
+### Architecture
+
+Nagios servers are standalone. All configuration of checks is via file.
+
+### Summary
+
+Nagios is suitable for basic monitoring of small and/or static systems where
+blackbox probing is sufficient.
+
+If you want to do whitebox monitoring, or have a dynamic or cloud based
+environment then Prometheus is a good choice.
+
+## Prometheus vs. Sensu
+
+[Sensu](https://sensuapp.org/) is broadly speaking a more modern Nagios.
+
+### Scope
+
+The same general scope differences as in the case of
+[Nagios](/docs/introduction/comparison/#prometheus-vs-nagios) apply here.
+
+The primary difference is that Sensu clients [register
+themselves](https://sensuapp.org/docs/0.27/reference/clients.html#what-is-a-sensu-client),
+and can determine the checks to run either from local or local configuration.
+Sensu does not have a limit on the amount of perfData.
+
+There is also a [client socket](https://sensuapp.org/docs/0.27/reference/clients.html#what-is-the-sensu-client-socket) permitting arbitrary check results to be pushed into Sensu.
+
+### Data model
+
+Sensu has the same rough data model as [Nagios](/docs/introduction/comparison/#prometheus-vs-nagios).
+
+### Storage
+
+Sensu has storage in Redis called stashes, this is used primarily for storing
+silences. It also stores all the clients that have registered with it.
+
+### Architecture
+
+Sensu has a [number of
+components](https://sensuapp.org/docs/0.27/overview/architecture.html). It uses
+RabbitMQ as a transport, Redis for current state and a separate Server for
```

Comment:
There's no Oxford commas throughout. I'd be a big fan of adding them everywhere though :P